src/main.zig

```diff
@@ -175,6 +175,15 @@ pub fn main() anyerror!void {
 }
 
 pub fn mainArgs(gpa: Allocator, arena: Allocator, args: []const []const u8) !void {
+    if (build_options.have_llvm) {
+        if (ZigClangIsLLVMUsingSeparateLibcxx()) {
```

Comment:
This is a clever trick to detect this problem, and I think we can work this into the codebase somehow. However, I'm not happy with it running unconditionally in main(), wasting CPU cycles during initialization. This adds up with many process invocations.

What do you think about the following compromise?
1. Move it to join the other LLVM initialization code in codegen/llvm.zig
2. Only include the check in debug builds of the compiler.

An alternate compromise idea would be to have this check happen on the `zig env` subcommand, since that command is already doing other kinds of introspection.

Reply:
That's a good point.

My biggest concern is that this bug is *very* subtle/misleading when it manifests, and I would like to avoid a maintainer tweaking the build system and accidentally ending up with a borked Zig. I'm sure plenty maintainers know to look out for this sort of thing, but it's still quite easy to do.

For that reason, I'm reluctant to make it debug-only

If the check is moved out of the "functional" path, I think it's a good idea to make sure it's at least likely to hit during the bug report process - so perhaps both `zig env` and `zig version`?

That seems likely to hit without being in the fast path